src/amp.erl

```diff
@@ -110,6 +111,13 @@ make_error_el(Errors, Rules) ->
                     {<<"code">>, error_code(hd(Errors))}],
            children = [ErrorMarker, RuleContainer]}.
 
+-spec update_rules(#xmlel{}, [amp_rule()]) -> #xmlel{}.
+update_rules(Packet = #xmlel{children = Children}, Rules) ->
+    Packet#xmlel{children = [case is_amp_el(El) of
+                                 true -> El#xmlel{children = [rule_to_xmlel(R) || R <- Rules]};
+                                 false -> El
+                             end || El <- Children]}.
```

Comment:
Can the case inside list comprehension be extracted to a function or a fun, I personally find code like the one less readable and harder to maintain.

Reply:
I somehow like to have such `case` expressions when the logic is plain and simple. I can change it though, it's no big deal for me.